### DISCUSSION TITLE
Call to a member function validate() on array

### DISCUSSION THREAD

**Discussion by harrisonratcliffe** [livewire/livewire#5150] [1. Help] [answered]
Hello, I am trying to submit a form and validate the data in another component's method. I'm getting an error and I'm not 100% sure what I'm doing wrong.

**Error:** `Call to a member function validate() on array` 

**Form component:**
```
public function submit()
{
    $this->emit('exampleEvent', $this->all());
}
```

**Component for validation:**
```
protected $listeners = ['exampleEvent' => 'validate'];

public function validate($data)
{
    $this->data = $data;

    $validator = $data>validate([
        'name' => ['required', 'string', 'max:50', 'min:4']
    ]);
```

If I do `dd($data);` I get this which looks all fine (I think?):
```
^ array:4 [▼
  "name" => "Example Name"
  "forceClose" => false
  "skipModals" => 0
  "destroySkipped" => false
]
```

Any help or suggestions is much appreciated!

**Comment by joshhanley** [comment, marked as answer]
@harrisonratcliffe you can't call `$data->validate()` as `$data` isn't an object which has a validate method, it is just an array of data.

If you want to manually validate data, you should use Laravel's validator to do this. See here for details https://laravel-livewire.com/docs/2.x/input-validation#custom-validators

Hope this helps!
